cfme/tests/automate/test_service_dialog.py

```diff
@@ -418,3 +422,60 @@ def test_dialog_items_default_values_on_different_screens():
         title: Test dialog items default values on different screens
     """
     pass
+
+
+@pytest.mark.meta(automates=[1571000])
+@pytest.mark.tier(3)
+@pytest.mark.customer_scenario
+@pytest.mark.parametrize(
+    "import_data",
+    [DatastoreImport("bz_1571000.zip", "bz_1571000", None)],
+    ids=["domain"]
+)
+@pytest.mark.parametrize("file_name", ["bz_1571000.yml"], ids=["dialog"])
+def test_automate_methods_from_dynamic_dialog_should_run_as_per_designed(
+    request, appliance, import_datastore, import_data, import_dialog, catalog
+):
+    """
+    Bugzilla:
+        1571000
+
+    Polarion:
+        assignee: nansari
+        casecomponent: Services
+        testtype: functional
+        initialEstimate: 1/16h
+        startsin: 5.9
+        tags: service
+    """
+    sd, ele_label = import_dialog
+    catalog_item = appliance.collections.catalog_items.create(
+        appliance.collections.catalog_items.GENERIC,
+        name=fauxfactory.gen_alphanumeric(),
+        description=fauxfactory.gen_alphanumeric(),
+        display_in=True,
+        catalog=catalog,
+        dialog=sd,
+    )
+    request.addfinalizer(catalog_item.delete_if_exists)
+    service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
+    patterns = [
+        ".*CC- dialog_instance1 value=.*",
+        ".*CC- dialog_instance2 value=.*",
+        ".*CC- dialog_instance3 value=.*",
+    ]
```

Comment:
@ganeshhubale I think we may need to do some more investigation here. Looking at the BZ, the problem was:
```
 method for Element A
  method for Element B
  method for Element C
  method for Element A
  method for Element B
  method for Element C```
was being logged.

The log validator in this case I believe would pass in this context since each pattern was hit. But really we need to be checking the count of those instances in the log. They should be logged only once each.  Let me know your thoughts here.

Reply:
Thanks for this suggestion :)
I have updated test case as below. Please let me know if we need more changes:
```
patterns = [
        ".*CC- dialog_instance1 value=.*",
        ".*CC- dialog_instance2 value=.*",
        ".*CC- dialog_instance3 value=.*",
    ]

    # Checking if automates method gets triggered by three instances for once after ordering catalog
    # item
    result = LogValidator(
        "/var/www/miq/vmdb/log/automation.log", matched_patterns=patterns
    )
    result.start_monitoring()
    view = navigate_to(service_catalogs, "Order")
    for i in range(len(patterns)):
        assert result.matches[patterns[i]] == 1
```